clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -652,6 +652,11 @@ void write(KafkaChannel channel) throws IOException {
             if (send != null) {
                 this.completedSends.add(send);
                 this.sensors.recordCompletedSend(nodeId, send.size(), currentTimeMs);
+
+                // To fix KAFKA-13559, i.e. select(timeout) blocks for 300 ms when the socket has no data but the buffer
+                // has data. Only happens when using SSL.
+                if (channel.hasBytesBuffered())
+                    madeReadProgressLastPoll = true;
```

Comment:
It's weird to change this read related variable after we have done a write.

Reply:
@ijuma Should I move this to `pollSelectionKeys` after calling `attemptWrite` (see below link)?

https://github.com/apache/kafka/blob/f2242f44fdee9b5575c8f359bacb0e5f457ffd42/clients/src/main/java/org/apache/kafka/common/network/Selector.java#L593

I have tested that this would also work.

Reply:
@ijuma I have moved the fix to `pollSelectionKeys` as I mentioned before to make it less weird (hopefully).

Reply:
PR updated.. so I'll resolve this comment..